h2o-core/src/main/java/water/rapids/Merge.java

```diff
@@ -125,7 +125,13 @@ public static Frame merge(final Frame leftFrame, final Frame riteFrame, final in
     final BigInteger riteBase = hasRite ? riteIndex._base [0] : leftBase;
 
     // initialize for double columns, may not be used....
-    long leftMSBfrom = riteBase.subtract(leftBase).shiftRight(leftShift).longValue();
+    boolean leftFrameEmpty = leftFrame.numRows()==0;
+    boolean riteFrameEmpty = riteFrame.numRows() == 0;
+    BigInteger bigZero = BigInteger.ZERO;
+    long leftMSBfrom = riteFrameEmpty? (leftFrameEmpty? bigZero.shiftRight(leftShift).longValue()
+                    : bigZero.shiftRight(leftShift).longValue()):
+            (leftFrameEmpty?riteBase.shiftRight(leftShift).longValue():
+                    riteBase.subtract(leftBase).shiftRight(leftShift).longValue());
     boolean riteBaseExceedsleftBase=riteBase.compareTo(leftBase)>0;
```

Comment:
leftMSBfrom/leftMSBto/riteMSBfrom/riteMSBto are all changed if there is empty frame involved.  If nothing is done, one of the values will be Long.MAX_VALUE and this will create the stall the user has reported.